app/models/user.rb

```diff
@@ -58,6 +58,14 @@ def self.from_oauth_hash(auth_hash)
     user
   end
 
+  def client_model
+    Proposal.client_model_for(self)
+  end
+
+  def client_model_slug
+    client_model.to_s.underscore.tr("/", "_")
```

Comment:
given that one of the business goals at the moment is to scale up to a large number of clients, I would prefer to avoid any kind of hard-coding of client-specific strings.